service/internal/proctelemetry/process_telemetry.go

```diff
@@ -127,6 +159,102 @@ func RegisterProcessMetrics(registry *metric.Registry, ballastSizeBytes uint64)
 	return nil
 }
 
+func (pm *processMetrics) recordWithOtel() error {
+	var err error
+
+	pm.otelProcessUptime, err = pm.meter.Float64ObservableCounter(
+		"process_uptime",
+		instrument.WithDescription("Uptime of the process"),
+		instrument.WithUnit(unit.Unit("s")))
+	if err != nil {
+		return err
+	}
+	_, err = pm.meter.RegisterCallback(func(_ context.Context, o otelmetric.Observer) error {
+		o.ObserveFloat64(pm.otelProcessUptime, pm.updateProcessUptime())
+		return nil
+	}, pm.otelProcessUptime)
+	if err != nil {
+		return err
+	}
+
+	pm.otelAllocMem, err = pm.meter.Int64ObservableGauge(
+		"process_runtime_heap_alloc_bytes",
+		instrument.WithDescription("Bytes of allocated heap objects (see 'go doc runtime.MemStats.HeapAlloc')"),
+		instrument.WithUnit(unit.Bytes))
+	if err != nil {
+		return err
+	}
+	_, err = pm.meter.RegisterCallback(func(_ context.Context, o otelmetric.Observer) error {
+		o.ObserveInt64(pm.otelAllocMem, pm.updateAllocMem())
+		return nil
+	}, pm.otelAllocMem)
+	if err != nil {
+		return err
+	}
+
+	pm.otelTotalAllocMem, err = pm.meter.Int64ObservableCounter(
+		"process_runtime_total_alloc_bytes",
+		instrument.WithDescription("Cumulative bytes allocated for heap objects (see 'go doc runtime.MemStats.TotalAlloc')"),
+		instrument.WithUnit(unit.Bytes))
+	if err != nil {
+		return err
+	}
+	_, err = pm.meter.RegisterCallback(func(_ context.Context, o otelmetric.Observer) error {
+		o.ObserveInt64(pm.otelTotalAllocMem, pm.updateTotalAllocMem())
+		return nil
+	}, pm.otelTotalAllocMem)
+	if err != nil {
+		return err
+	}
+
+	pm.otelSysMem, err = pm.meter.Int64ObservableGauge(
+		"process_runtime_total_sys_memory_bytes",
+		instrument.WithDescription("Total bytes of memory obtained from the OS (see 'go doc runtime.MemStats.Sys')"),
+		instrument.WithUnit(unit.Bytes))
+	if err != nil {
+		return err
+	}
+	_, err = pm.meter.RegisterCallback(func(_ context.Context, o otelmetric.Observer) error {
```

Comment:
Why not using https://pkg.go.dev/go.opentelemetry.io/otel/metric@v0.35.0/instrument#WithInt64Callback in the previous call.

Reply:
Didn't notice this option, but agreed this makes it cleaner. Thanks for pointing it out!